### PR TITLE
Improvements and new features

### PR DESCRIPTION
**Pull request by miiu96**
- Added a new endpoint `/simulator/add-keys` to add in the simulator validator keys ( the keys will be added in the multi-key handler )
- Changed the version of `mx-chain-go`
- Fixed the problem when the simulator gets stuck ( the `log.WARN` with the message `shard is stuck` should no longer appear)
- Added in the `config.toml` file a parameter to set the initial round of the simulator
- Extended the Readme file
- Removed the **3 seconds delay** from the epoch change mechanism

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -174,7 +174,42 @@ Example:
...
+    "ZjVkYjgwZDE1NzE5MDJiN2UzMDNlNDIzOTUzZGU2NTQ4NzBiYzM1MDhmMThkNGRhODgzODk1NTI3ZjcyMjYxYw==",
+    "ZTVhYTU0NjI0ZmRjNDZkMDdmNDU5ZGZiZDFmNmUxYWZlMTRmN2YyOTY1ZTJiMGJhZjBmMGE0MGQ3ZjYwNDYxYg==",
+  ]
+}
+```
+
+
+##### Response
+- **Status Codes:**
+  - `200 OK`: Validator keys was added successfully.
```
Comment (sstanculeanu) [inline README.md:202]:
```suggestion
  - `200 OK`: Validator keys were added successfully.
```

File: cmd/chainsimulator/main.go [hunk 2]
```diff
@@ -137,9 +137,15 @@ func startChainSimulator(ctx *cli.Context) error {
 
 	startTimeUnix := ctx.GlobalInt64(startTime.Name)
 	apiConfigurator := api.NewFreePortAPIConfigurator("localhost")
+
+	tempDir := os.TempDir()
+	defer func() {
+		_ = os.RemoveAll(tempDir)
```
Comment (sstanculeanu) [inline cmd/chainsimulator/main.go:143]: is it ok to remove all in case it returns /tmp ?

File: pkg/proxy/api/endpoints.go [hunk 3]
```diff
@@ -112,3 +115,21 @@ func (ep *endpointsProcessor) setStateMultiple(c *gin.Context) {
 
 	shared.RespondWith(c, http.StatusOK, gin.H{}, "", data.ReturnCodeSuccess)
 }
+
+func (ep *endpointsProcessor) addValidatorKeys(c *gin.Context) {
+	validatorsKeys := &dtosc.ValidatorKeys{}
+
+	err := c.ShouldBindJSON(&validatorsKeys)
```
Comment (sstanculeanu) [inline pkg/proxy/api/endpoints.go:122]:
```suggestion
	err := c.ShouldBindJSON(validatorsKeys)
```

File: cmd/chainsimulator/main.go (hunk 2 above)
Comment (iulianpascalau) [reply cmd/chainsimulator/main.go:143]: Good catch @sstanculeanu :+1: 
maybe use something from this example?
```GO
aaa, err := os.MkdirTemp(os.TempDir(), "chainSimulator")
assert.Nil(t, err)

fmt.Println(aaa)
```

File: README.md (hunk 1 above)
Comment (miiu96) [reply README.md:202]: done

File: cmd/chainsimulator/main.go (hunk 2 above)
Comment (miiu96) [reply cmd/chainsimulator/main.go:143]: done

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**